.github/workflows/ci-build.yml

```diff
@@ -9,12 +9,32 @@ on:
 jobs:
   janus_build:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: "write"
+      contents: "read"
     env:
       CARGO_INCREMENTAL: 0
       CARGO_TERM_COLOR: always
+      DAPHNE_INTEROP_CONTAINER: prebuilt=${{ secrets.DAPHNE_PREBUILT_IMAGE_NAME_AND_TAG }}
       RUSTFLAGS: "-D warnings"
     steps:
     - uses: actions/checkout@v3
+    # See https://github.com/google-github-actions/auth#authenticating-to-container-registry-and-artifact-registry
+    - id: "gcp-auth"
+      name: "Authenticate to GCP"
+      uses: "google-github-actions/auth@v0"
+      with:
+        workload_identity_provider: ${{ secrets.GCP_ARTIFACT_PUBISHER_WORKLOAD_IDENTITY_PROVIDER }}
+        service_account: ${{ secrets.GCP_ARTIFACT_PUBLISHER_DEPLOY_SERVICE_ACCOUNT }}
```

Comment:
Ah, this will need to change to the read-only service account, to line up with the change in divviup/janus-ops#262.

Reply:
Thanks! I put this down on Friday afternoon because I needed to look at some DAP changes and you fixed it before I could pick this back up.